include/range/v3/view/sliding.hpp

```diff
@@ -0,0 +1,269 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Tobias Mayer 2016
+//  Copyright Casey Carter 2016
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_SLIDING_HPP
+#define RANGES_V3_VIEW_SLIDING_HPP
+
+#include <utility>
+#include <functional>
+#include <meta/meta.hpp>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/begin_end.hpp>
+#include <range/v3/range_traits.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/detail/optional.hpp>
+#include <range/v3/view_adaptor.hpp>
+#include <range/v3/view/all.hpp>
+#include <range/v3/view/counted.hpp>
+#include <range/v3/view/view.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-views
+        /// @{
+        template<typename Rng>
+        struct sliding_view
+          : view_adaptor<
+                sliding_view<Rng>,
+                Rng,
+                is_finite<Rng>::value ? finite : range_cardinality<Rng>::value>
+          , private detail::non_propagating_cache<
+                range_iterator_t<Rng>,
+                sliding_view<Rng>,
+                !(RandomAccessRange<Rng>() && SizedRange<Rng>())>
+        {
+        private:
+            CONCEPT_ASSERT(ForwardRange<Rng>());
+            friend range_access;
+            struct adaptor;
+
+            static constexpr bool cache_enabled =
+                !(RandomAccessRange<Rng>() && SizedRange<Rng>());
```

Comment:
I have it from Andrew that class static members are slow to instantiate. An alias or even a static member function is lighter weight.